backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryHttpClient.java

```diff
@@ -124,10 +125,19 @@ private void sendDelete(TelemetryPayload payload) {
   }
 
   private void sendPost(TelemetryPayload payload) {
+    if (isDebugTelemetryEnabled()) {
+      LOG.info("Sending telemetry payload.");
+      LOG.info(payload.toJson());
+    }
     try (var response = client.post(endpoint, HttpClient.JSON_CONTENT_TYPE, payload.toJson())) {
       if (!response.isSuccessful() && InternalDebug.isEnabled()) {
         LOG.error("Failed to upload telemetry data: {}", response.toString());
       }
     }
   }
+
+  @VisibleForTesting
+  boolean isDebugTelemetryEnabled(){
```

Comment:
How about `isTelemetryLogEnabled` for the method name and `SONARLINT_TELEMETRY_LOG` for the env var name?

Reply:
Addressed as requested